sympy/printing/tests/test_glsl.py

```diff
@@ -399,6 +402,95 @@ def test_Matrices_1x7():
     assert gl(A) == 'float[7](1, 2, 3, 4, 5, 6, 7)'
     assert gl(A.transpose()) == 'float[7](1, 2, 3, 4, 5, 6, 7)'
 
+def test_Matrices_1x7_array_type_int():
+    gl = glsl_code
+    A = Matrix([1,2,3,4,5,6,7])
+    assert gl(A, array_type='int') == 'int[7](1, 2, 3, 4, 5, 6, 7)'
+
+def test_Tuple_array_type_custom():
+    gl = glsl_code
+    A = symbols('a b c')
+    assert gl(A, array_type='AbcType', glsl_types=False) == 'AbcType[3](a, b, c)'
+
+def test_Matrices_1x7_spread_assign_to_symbols():
+    gl = glsl_code
+    A = Matrix([1,2,3,4,5,6,7])
+    assign_to = symbols('x.a x.b x.c x.d x.e x.f x.g')
+    assert gl(A, assign_to=assign_to) == textwrap.dedent('''\
+        x.a = 1;
+        x.b = 2;
+        x.c = 3;
+        x.d = 4;
+        x.e = 5;
+        x.f = 6;
+        x.g = 7;'''
+    )
+
+def test_spread_assign_to_nested_symbols():
+    gl = glsl_code
+    expr = ((1,2,3), (1,2,3))
+    assign_to = (symbols('a b c'), symbols('x y z'))
+    assert gl(expr, assign_to=assign_to) == textwrap.dedent('''\
+      a = 1;
+      b = 2;
+      c = 3;
+      x = 1;
+      y = 2;
+      z = 3;'''
+   )
+
+def test_spread_assign_to_deeply_nested_symbols():
+    gl = glsl_code
+    a, b, c, x, y, z = symbols('a b c x y z')
+    expr = (((1,2),3), ((1,2),3))
+    assign_to = (((a, b), c), ((x, y), z))
+    assert gl(expr, assign_to=assign_to) == textwrap.dedent('''\
+      a = 1;
+      b = 2;
+      c = 3;
+      x = 1;
+      y = 2;
+      z = 3;'''
+   )
+
+def test_matrix_of_tuples_spread_assign_to_symbols():
+   gl = glsl_code
+   expr = Matrix([[(1,2),(3,4)],[(5,6),(7,8)]])
+   assign_to = (symbols('a b'), symbols('c d'), symbols('e f'), symbols('g h'))
+   assert gl(expr, assign_to) == textwrap.dedent('''\
+      a = 1;
+      b = 2;
+      c = 3;
+      d = 4;
+      e = 5;
+      f = 6;
+      g = 7;
+      h = 8;'''
+)
```

Comment:
Indents should be multiples of four space - here I see 6, 7, and 8-space indents.